Examples/Algorithms/Geant4/src/DetectorConstructionHelper.cpp

```diff
@@ -0,0 +1,37 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Geant4/DetectorConstructionHelper.hpp"
+
+#include "ActsExamples/TelescopeDetector/TelescopeDetector.hpp"
+#include "ActsExamples/TelescopeDetector/TelescopeG4DetectorConstruction.hpp"
+#ifdef ACTS_GEANT4_DD4HEP
```

Comment:
I am not too happy about those. any other ideas @benjaminhuth @paulgessinger ?

Reply:
Might be cleaner to determine this at CMake level only and add different `.cpp` files to the target? That's ~ what I did for the python bindings.

Reply:
One alternative that comes to my mind would be to implement this in a pure python function, that can dynamically check if a type is available or not... Might also be an option, because I don't think we need to call this ever directly from C++, right?

Besides this I'm not sure if handling in CMake is useful here, because thant we could not do it in a single function, right?